src/providers/treeViewProvider.ts

```diff
@@ -0,0 +1,82 @@
+import * as vscode from 'vscode';
+import { TreeItemCollapsibleState } from 'vscode';
+import { SelectedRequest, Selector } from '../utils/selector';
+
+export class HttpTreeProvider implements vscode.TreeDataProvider<HttpClientItem> {
+
+    private _onDidChangeTreeData: vscode.EventEmitter<HttpClientItem | undefined> = new vscode.EventEmitter<HttpClientItem | undefined>();
+    readonly onDidChangeTreeData: vscode.Event<HttpClientItem | undefined> = this._onDidChangeTreeData.event;
+
+    constructor() {
+    }
+
+    refresh(): void {
+        this._onDidChangeTreeData.fire();
+    }
+
+    getTreeItem(element: HttpClientItem): vscode.TreeItem {
+        return element;
+    }
+
+    getChildren(element?: HttpClientItem): Thenable<HttpClientItem[]> {
+        return Promise.resolve(this.getHttpClientItems(element));
+    }
+
+    private getHttpClientItems(element): Thenable<HttpClientItem[]> {
+        if (element) {
```

Comment:
I prefer the design that the `HttpClientItems` holds the range of each request, instead of the parsed ones for the following reasons:

- If you have the range, you don't need to write the method `runRequest`, maybe we can reuse the original `run` method
- If we have range, we can have the capability to click the request in the explorer, and we can go to the actual range of the request

Reply:
I tried that first, but the run method takes the current editor while the requests from the tree doesn't have to have any editor opened since they already has the document. So a change in the run is necessary anyway. I also tried to pass the document to the run but then the code becomes too complex.
What do you think? any suggestions?